internal/driver/daemon.go

```diff
@@ -84,32 +84,42 @@ func (r *RegistryDefault) ServeWrite(ctx context.Context) func() error {
 	}
 }
 
+func (r *RegistryDefault) ServeMetrics(ctx context.Context) func() error {
+	rt := r.MetricsRouter()
+
+	return func() error {
+		return multiplexPortNoGRPC(ctx, r.Config().MetricsListenOn(), rt)
+	}
```

Comment:
In cases were you don't have gRPC, you don't have to multiplex at all.

```suggestion
	return func() error {
		graceful.WithDefaults(&http.Server{
			Handler: r.MetricsRouter(),
			Addr:    r.Config().MetricsListenOn(),
		})
	}
```

Will make the code a lot less complex.